src/opm/output/eclipse/AggregateUDQData.cpp

```diff
@@ -64,30 +64,423 @@ namespace {
+            Opm::UDQTokenType::binary_op_uadd,
+            Opm::UDQTokenType::binary_op_umul,
+            Opm::UDQTokenType::binary_op_umin,
+            Opm::UDQTokenType::binary_op_umax
+        };
+        for (const auto& tok_type : type_1) {
+            if (token == tok_type) {
+                type = true;
+                break;
+            }
+        }
+        return type;
+    }
+
+        // function to return true if token is a binary operator: type
+    bool tokenTypeParen(const Opm::UDQTokenType& token) {
+        bool type = false;
+        std::vector <Opm::UDQTokenType> type_1 = {
+            Opm::UDQTokenType::open_paren,
+            Opm::UDQTokenType::close_paren,
+        };
+        for (const auto& tok_type : type_1) {
+            if (token == tok_type) {
+                type = true;
+                break;
+            }
+        }
+        return type;
+    }
+
+    // function to return true if token is a binary operator: anytype
+    bool tokenTypeBinaryOp(const Opm::UDQTokenType& token) {
+        bool type = false;
+        std::vector <Opm::UDQTokenType> type_1 = {
+            Opm::UDQTokenType::binary_op_add,
+            Opm::UDQTokenType::binary_op_sub,
+            Opm::UDQTokenType::binary_op_div,
+            Opm::UDQTokenType::binary_op_mul,
+            Opm::UDQTokenType::binary_op_pow,
+            Opm::UDQTokenType::binary_op_uadd,
+            Opm::UDQTokenType::binary_op_umul,
+            Opm::UDQTokenType::binary_op_umin,
+            Opm::UDQTokenType::binary_op_umax,
+            Opm::UDQTokenType::binary_cmp_eq,
+            Opm::UDQTokenType::binary_cmp_ne,
+            Opm::UDQTokenType::binary_cmp_le,
+            Opm::UDQTokenType::binary_cmp_ge,
+            Opm::UDQTokenType::binary_cmp_lt,
+            Opm::UDQTokenType::binary_cmp_gt
 
-        int num_type_1 = 0;
+        };
         for (const auto& tok_type : type_1) {
-            num_type_1 += tokens.count(tok_type);
+            if (token == tok_type) {
+                type = true;
+                break;
+            }
         }
-        type = (num_type_1 > 0) ? -1 : -4;
         return type;
     }
 
+    // function to return the precedence of the current operator/function
+    bool operatorToken(const Opm::UDQTokenType& token) {
+        bool opTok = false;
+        if (tokenTypeFunc(token)) opTok = true;
+        if (tokenTypeBinaryCmpOp(token)) opTok = true;
+        if (tokenTypeBinaryPowOp(token)) opTok = true;
+        if (tokenTypeBinaryMulDivOp(token)) opTok = true;
+        if (tokenTypeBinaryAddSubOp(token)) opTok = true;
+        if (tokenTypeBinaryUnionOp(token)) opTok = true;
+        return opTok;
+    }
+
+        // function to return index number of last binary token not inside bracket that is ending the expression
+    int noOperators(const std::vector<Opm::UDQToken>& modTokens) {
+        int noOp = 0;
+        for (std::size_t ti = 0; ti < modTokens.size(); ti++) {
```

Comment:
Here you can just use range based for and no index.
```
for (const auto& modToken : modTokens) {
     if (operatorToken(modToken) || ....)
}
```

Reply:
OK  - changed as suggested.